src/native-common/Image.tsx

```diff
@@ -121,7 +122,7 @@ export class Image extends React.Component<Types.ImageProps, ImageState> impleme
         const sourceOrHeaderChanged = (nextProps.source !== this.props.source ||
```

Comment:
fyi: [componentWillReceiveProps](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops) will be deprecated in the future. 

> Using this lifecycle method often leads to bugs and inconsistencies, and for that reason it is going to be deprecated in the future.